File: Tutorial4-7LayerLibrary/src/main/java/com/test/tutorial4_7layerlibrary/di/BaseAppModule.java
package com.test.tutorial4_7layerlibrary.di;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import com.test.tutorial4_7layerlibrary.model.ToastMaker;
import com.test.tutorial4_7layerlibrary.model.User;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;


@Module
public class BaseAppModule {

    @Singleton
    @Provides
    SharedPreferences provideSharedPreferences(Application application) {
        return application.getSharedPreferences("PrefName", Context.MODE_PRIVATE);
    }

    @Singleton
    @Provides
    User provideUser() {
        return  new User("Unknown","abc@xyz.com");
    }

    @Singleton
    @Provides
    ToastMaker provideToastMaker(Application application) {
        return new ToastMaker(application);
    }

}

File: Tutorial4-7LayerLibrary/src/main/java/com/test/tutorial4_7layerlibrary/di/BaseAppModule.kt
package com.test.tutorial4_7layerlibrary.di

import android.app.Application
import android.content.Context
import android.content.SharedPreferences
import com.test.tutorial4_7layerlibrary.model.ToastMaker
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class BaseAppModule {

    @Singleton
    @Provides
    fun provideSharedPreferences(application: Application): SharedPreferences =
            application.getSharedPreferences("PrefName", Context.MODE_PRIVATE)

    @Singleton
    @Provides
    fun provideToastMaker(application: Application) = ToastMaker(application)
}